lib/compute/address.js

```diff
@@ -0,0 +1,129 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*!
+ * @module compute/address
+ */
+
+'use strict';
+
+/**
+ * @type {module:common/util}
+ * @private
+ */
+var util = require('../common/util.js');
+
+/*! Developer Documentation
+ *
+ * @param {module:region} region - Region this address belongs to.
+ * @param {string} name - The name of the address.
+ */
+/**
+ * An Address object allows you to interact with a Google Compute Engine
+ * address.
+ *
+ * @constructor
+ * @alias module:compute/address
+ *
+ * @example
+ * var gcloud = require('gcloud')({
+ *   keyFilename: '/path/to/keyfile.json',
+ *   projectId: 'grape-spaceship-123'
+ * });
+ *
+ * var compute = gcloud.compute();
+ *
+ * var region = compute.region('region-name');
+ *
+ * var address = region.address('address1');
+ */
+function Address(region, name) {
+  this.region = region;
+  this.name = name;
+}
+
+/**
+ * Delete the address.
+ *
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * address.delete(function(err, operation) {
+ *   // `operation` is an Operation object that can be used to check the status
+ *   // of the request.
+ * });
+ */
+Address.prototype.delete = function(callback) {
+  callback = callback || util.noop;
+
+  var region = this.region;
+
+  this.makeReq_('DELETE', '', null, null, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    var operation = region.operation(resp.name);
+    operation.metadata = resp;
+
+    callback(null, operation, resp);
+  });
+};
+
+/**
+ * Get the metadata of this address.
+ *
+ * @param {function=} callback - The callback function.
+ *
+ * @example
+ * address.getMetadata(function(err, metadata, apiResponse) {});
+ */
+Address.prototype.getMetadata = function(callback) {
+  callback = callback || util.noop;
+
+  var self = this;
+
+  this.makeReq_('GET', '', null, null, function(err, resp) {
+    if (err) {
+      callback(err, null, resp);
+      return;
+    }
+
+    self.metadata = resp;
+
+    callback(null, self.metadata, resp);
+  });
+};
+
+/**
+ * Make a new request object from the provided arguments and wrap the callback
+ * to intercept non-successful responses.
+ *
+ * @private
+ *
+ * @param {string} method - Action.
+ * @param {string} path - Request path.
+ * @param {*} query - Request query object.
+ * @param {*} body - Request body contents.
+ * @param {function} callback - The callback function.
+ */
+Address.prototype.makeReq_ = function(method, path, query, body, callback) {
```

Comment:
Gotcha, I was just digging to find out what our "preferred" approach was! (And now I know!)